### PR TITLE
Remember last route type

### PR DESCRIPTION
**Pull request by itmuckel**
In the Add Tracks/Busses dialog the last chosen route type will be remembered now. It works only for the builtin types.
Also it works only for ardour as it looked to me as if there were some default selections made for Mixbus.

The other two commits remove some unused imports and unimplemented methods and can be dropped if you don't want them.

### CODE REVIEW

File: gtk2_ardour/add_route_dialog.cc [hunk 1]
```diff
@@ -598,11 +591,31 @@ AddRouteDialog::type_wanted()
 	} else if (str == _("Foldback Busses")) {
 		return FoldbackBus;
 	} else {
-		assert (0);
```
Comment (itmuckel) [inline gtk2_ardour/add_route_dialog.cc:601]: I ran into problems when selecting a non-builtin-type and trying to save/load it. I defaulted to AudioTrack, but maybe this is not what we want.

File: gtk2_ardour/add_route_dialog.cc [hunk 2]
```diff
@@ -598,11 +591,31 @@ AddRouteDialog::type_wanted()
...
+			return _("Audio Tracks");
+		case VCAMaster:
+			return _("VCA Masters");
+		case FoldbackBus:
+			return _("Foldback Busses");
+		default:
+			return _("Audio Tracks");
+			break;
+	}
+}
```
Comment (x42) [inline gtk2_ardour/add_route_dialog.cc:617]: These should probably not be translatable. In general the state should not contain translatable strings. We use `X_("..")` to indicate that.

**Comment by x42** [comment]
The mixbus special case can go away. We don't ship the "Create Audio Tracks Interactively" script anymore.

File: gtk2_ardour/add_route_dialog.h [hunk 3]
```diff
@@ -59,7 +59,7 @@ class AddRouteDialog : public ArdourDialog
 {
 public:
 	AddRouteDialog ();
-	~AddRouteDialog ();
+	~AddRouteDialog () override;
```
Comment (x42) [inline gtk2_ardour/add_route_dialog.h:62]: Is there a reason to keep this? The d'tor does nothing, so we should just remove it. Fewer LOC is better :)

**Comment by x42** [comment]
All in all. This is a very really nice cleanup and well done feature.
I don't yet see an easy way to make the state untranslatable. 
Unless you have a good idea, I'm tempted to just merge it and sort this out in a follow up commit.

**Comment by pauldavisthefirst** [comment]
The usual way to put enum types into untranslatable state strings is to register the enum in the appropriate enums.cc file, then use enum_2_string()

File: gtk2_ardour/add_route_dialog.h [hunk 4]
```diff
@@ -47,6 +47,7 @@
 #include "ardour/plugin.h"
 #include "ardour/types.h"
 #include "ardour/template_utils.h"
+#include "ardour/route_group.h"
```
Comment (itmuckel) [inline gtk2_ardour/add_route_dialog.h:50]: I needed this import otherwise it wouldn't compile. Don't know why.

**Comment by x42** [comment]
Rebased and merged as 7.0-pre0-2714 2a4ebd17d6
Thank you very much!

**Pull request closed without merging**